.github/scripts/transform_dbt_test_results.py

```diff
@@ -882,15 +901,20 @@ def create(
             git_author=git_author,
         )
 
+    def to_dict(self) -> typing.Dict:
+        return dataclasses.asdict(self)
```

Comment:
We need to extend `dataclasses.asdict()` with extra serialization steps for `TestRunFailingRowMetadata` (see below) so this stub function is intended to unify the interface across `Metadata` classes. I considered a class mixin to reduce some of the duplication between `TestRunMetadata` and `TestRunResultMetadata`, both of which implement this function in the same way, but I figure that unless/until we extend the `Metadata` interface further it's probably not worth the extra complexity that a mixin class incurs.